### PR TITLE
Python: Add code conventions for Python

### PR DESCRIPTION
**Pull request by Fokko**
Since we're approaching the first version of the PyIceberg library, I think it would be good to formalize the code conventions a bit. Would love to hear what you think of this.

### CODE REVIEW

File: python/CONTRIBUTING.md [hunk 1]
```diff
@@ -94,3 +94,31 @@ make test PYTEST_ARGS="--pdb"
...
+## API Compatibility
+
+We try to keep the Python public API compatible across versions. The Python official [PEP-8](https://peps.python.org/pep-0008/) defines Public methods as: _Public attributes should have no leading underscores_. This means not removing any methods without any notice, or removing or renaming any existing parameters. Adding new optional parameters is okay.
+
+If you want to remove a method, please add a deprecation notice by annotating the function using `@deprecated`:
+
+```python
+from pyiceberg.utils.deprecated import deprecated
+
+@deprecated(deprecated_in="0.1.0", removed_in="0.2.0")
```
Comment (rdblue) [inline python/CONTRIBUTING.md:111]: We may want to add a way to specify what to use instead, but this is great!

File: python/CONTRIBUTING.md [hunk 2]
```diff
@@ -94,3 +94,31 @@ make test PYTEST_ARGS="--pdb"
...
+def load_something():
+    pass
+```
+
+Which will return:
```
Comment (rdblue) [inline python/CONTRIBUTING.md:116]: What about "warn" rather than "return"?

**Review by rdblue — APPROVED** [review]
Looks great! Merge when you're ready and tests are passing.

File: python/CONTRIBUTING.md (hunk 1 above)
Comment (Fokko) [reply python/CONTRIBUTING.md:111]: Great suggestion, added it. Thanks!

File: python/CONTRIBUTING.md (hunk 2 above)
Comment (Fokko) [reply python/CONTRIBUTING.md:116]: Good one, thanks!

**Pull request merged**